ACPs/118-warp-signature-request/README.md

```diff
@@ -0,0 +1,100 @@
+
+Standardizing the Warp Signature Request interface by defining it as a format for `AppRequest` message payloads in AvalancheGo would simplify the implementation of ACP-75, and streamline signature aggregation for out-of-protocol services such as Warp message relayers.
+
+## Specification
+
+We propose the following types, implemented as Protobuf types that may be decoded from the `AppRequest`/`AppResponse` `app_bytes` field. By way of example, this approach is currently used to [implement](https://github.com/ava-labs/avalanchego/blob/v1.11.10-status-removal/proto/sdk/sdk.proto#7) and [parse](https://github.com/ava-labs/avalanchego/blob/v1.11.10-status-removal/network/p2p/gossip/message.go#22) gossip `AppRequest` types.
+
+- `SignatureRequest` includes two fields, both of which are optional. `data` specifies the payload that the returned signature should correspond to. In other words, `data` should be a serialized unsigned Warp message for which the requester is requesting a signature. `justification` specifies arbitrary data that the requested node may use to decide whether or not `data` is a payload it is willing to sign.
+
+    ```protobuf
+    message SignatureRequest struct {
+        bytes data = 1;
+        bytes justification = 2;
+    }
+    ```
+
+- `SignatureResponse` is the corresponding `AppResponse` type that returns the requested signature. 
+
+    ```protobuf
+    message SignatureResponse {
+        bytes signature = 1;
+    }
+    ```
+
+For each of these types, VMs must implement corresponding `AppRequest` and `AppResponse` handlers to adhere to the interface.
+
+## Use Cases
+
+### Sign a Warp Message
+
+`SignatureRequest` can be used to request a signature over a Warp message by serializing the unsigned Warp message into the `data` field, and omitting the `justification` field.
+
+### Sign a known Warp Message given its ID
+
+Subnet EVM and Coreth currently handle signature requests for known Warp messages by implementing the following `AppRequest` message type:
+
+```
+type MessageSignatureRequest struct {
+	MessageID ids.ID
+}
+```
+
+For all Warp messages that the node has seen, (i.e. on-chain message sent through the [Warp Precompile](https://github.com/ava-labs/subnet-evm/tree/v0.6.7/precompile/contracts/warp) and [off-chain](https://github.com/ava-labs/subnet-evm/blob/v0.6.7/plugin/evm/config.go#L226) Warp messages) there exists a mapping of Warp message ID to the full unsigned message bytes. Subnet EVM and Coreth nodes are therefore able to retreive and sign the full Warp message given its message ID.
+
+`SignatureRequest` could be used for this case by specifying the Warp message ID as the `justification` field, and omitting the `data` field.
```

Comment:
I guess it's up to the implementations either way, but I find it a bit confusing that the `data` field is considered optional since it's otherwise thought of as the data to be signed. 

I was picturing a case where the raw data to be signed is not included in the request as a separate interface, not part of this standard.

Reply:
Reworded the specification such that `data` is not optional, and removed the "request a signature by message ID" use case.